src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1569,6 +1574,17 @@ void TorrentImpl::fileSearchFinished(const Path &savePath, const PathList &fileN
     endReceivedMetadataHandling(savePath, fileNames);
 }
 
+void TorrentImpl::updateHealthStatus()
+{
+    const bool hasNoWorkingTracker = std::all_of(m_trackerEntries.cbegin(), m_trackerEntries.cend()
+            , [](const TrackerEntry &trackerEntry)
+    {
+        return (trackerEntry.status == TrackerEntry::Status::NotWorking);
+    });
+
+    m_isDead = hasNoWorkingTracker && isDHTDisabled() && isPEXDisabled() && isLSDDisabled();
```

Comment:
Should a dead torrent be define as such by the user's choice of options? I think not.